Adding Deploy to Node.js to AWS Markdown

Jazli14 opened the pull request:
Initial draft for markdown explaining how to deploy a Node.js application to AWS using Docker containerization.

Umair-H reviewed (changes requested):
Seems like a good blueprint to your article.

I think you can try to lay out some steps that the person would take to go about deploying the Node.js app through Docker on AWS. Be sure to list the steps out in order and be explicit about them.

Jazli14 commented:
> Seems like a good blueprint to your article.
> 
> I think you can try to lay out some steps that the person would take to go about deploying the Node.js app through Docker on AWS. Be sure to list the steps out in order and be explicit about them.

Thanks for the great feedback! I'll get to right to work on those additions.

Umair-H reviewed (changes requested):
Nice work implementing my feedback, some other things I think might make your article better are to add more specifics on each part, for example on how to actually create a task definition and how to configure the ECS. Add another part where you have to make it exposed to any IP (IPV4 or IPV6). This way your article comes off as more concrete with all the details attached.

Jazli14 commented:
> Nice work implementing my feedback, some other things I think might make your article better are to add more specifics on each part, for example on how to actually create a task definition and how to configure the ECS. Add another part where you have to make it exposed to any IP (IPV4 or IPV6). This way your article comes off as more concrete with all the details attached.

Hey Umair, thanks for the quick response.

I think I have added all of the new additions you mentioned including more detailed steps to deploy the Node.js application. I added the step to make it exposed to any IP and more specifics around creating the task definition, cluster and container repository. Let me know if you have any other things I should fix as well, and I'll try get on that

Umair-H reviewed (changes requested):
Great additions I think this will help provide students with that extra information that is necessary when learning this topic. 

I think shoukd add a table of contents to allow users to readily access information they need. Try to include some links to some external articles that help them to set up other AWS services like databases through RDS or how to use an S3 bucket with Node.js. Maybe also include a rough diagram image on what is happening.

Jazli14 commented:
> Great additions I think this will help provide students with that extra information that is necessary when learning this topic.
> 
> I think shoukd add a table of contents to allow users to readily access information they need. Try to include some links to some external articles that help them to set up other AWS services like databases through RDS or how to use an S3 bucket with Node.js. Maybe also include a rough diagram image on what is happening.

Thanks for the advice Umair. These sound like great additions to put in my article.

I have now added the table of contents to help navigate to whatever the reader's desired information they want. I also added some other external links that would better integrate AWS services into their Node.js application which you mentioned in your feedback.

Umair-H reviewed (changes requested):
I like how you incorporated the table of contents to let the user readily access whatever content they need. I also liked the new additions of the external links to learn how to incorporate S3 buckets and how to utilize an RDS database with Node.js. 

Everything seems great except for that diagram I was talking about or some sort of visual to help the reader. I think it will really help nail any confusions and provide an overview of what you're talking about. 

Other than that great work!

Jazli14 commented:
> I like how you incorporated the table of contents to let the user readily access whatever content they need. I also liked the new additions of the external links to learn how to incorporate S3 buckets and how to utilize an RDS database with Node.js.
> 
> Everything seems great except for that diagram I was talking about or some sort of visual to help the reader. I think it will really help nail any confusions and provide an overview of what you're talking about.
> 
> Other than that great work!

Hey Umair, thanks for checking up on the work.

I finally made a diagram for the deployment process which includes containerizing Node.js app, building the image and pushing to Amazon ECR repo. And I made another diagram to demonstrate how it is deployed by using ECS which implements it on an EC2 instance.

I also added an extra link which provides a much more detailed version of my article if the reader needs it.

Umair-H reviewed (approved):
Looking good! Your addition of diagrams should make it much easier for new students to navigate through your article. The links also look good, for people looking for extra info on the topic. Overall, awesome work incorporating all the previous changes—I'm happy with how everything looks!
